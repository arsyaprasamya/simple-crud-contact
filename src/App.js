import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import { Switch, Route, withRouter } from 'react-router-dom';

import Home from './components/Home';
import NewContact from './components/NewContact';
import EditContact from './components/EditContact';

import GlobalStyle from './components/styles/Global';

const Container = styled.div`
  width: 100%;
  max-width: 41rem;
  margin: 0 auto;
`;

const Heading = styled.h1`
  font-family: inherit;
  font-size: 5rem;
  color: #2c3e50;
  text-align: center;
  padding: 4rem 0 10rem;
`;

class App extends Component {
  render() {
    return (
      <Fragment>
        <Container>
          <Heading>MY CONTACT</Heading>
          <Switch>
            <Route path='/edit/:id' component={EditContact} />
            <Route path='/new' component={NewContact} />
            <Route path='/' component={Home} />
          </Switch>
        </Container>
        <GlobalStyle />
      </Fragment>
    );
  }
}

export default withRouter(App);
