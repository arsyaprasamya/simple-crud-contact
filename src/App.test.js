import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  const wrapper = mount(<App />);
  expect(wrapper.state("error")).toEqual(null);
});

it("renders without crashing", () => {
  shallow(<App />);
});
it("renders Account header", () => {
  const wrapper = shallow(<App />);
  const header = <h1>Display Active Users Account Details</h1>;
  // expect(wrapper.contains(welcome)).toBe(true);
  expect(wrapper.contains(header)).toEqual(true);
});

it.skip("renders without crashing", () => {
  shallow(<App />);
});

it("renders Account header", () => {
  const wrapper = shallow(<App />);
  const header = <h1>Display Active Users Account Details</h1>;
  expect(wrapper.contains(header)).toEqual(true);
});

it("renders correctly", () => {
  const tree = shallow(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});
