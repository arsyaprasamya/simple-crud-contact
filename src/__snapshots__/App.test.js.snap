exports[`renders correctly 1`] = `
<Fragment>
  <h1>
    Display Active Users Account Details
  </h1>
  <h3>
    Fetching Users...
  </h3>
</Fragment>
`;